scripts/take_screenshots.py

```
@@ -0,0 +1,97 @@
+    if not isinstance(shots, list):
+        print(f"error: {args.config} must contain a YAML list of shots", file=sys.stderr)
+        return 2
+
+    failures = []
+    for index, shot in enumerate(shots, start=1):
+        output = shot.get("output", "?")
+        Path(output).parent.mkdir(parents=True, exist_ok=True)
+        print(f"[{index}/{len(shots)}] {output}", flush=True)
+        try:
+            # argv is a fixed list from shots.yml, never shell-interpreted
+            result = subprocess.run(build_command(shot, args.timeout), check=False)  # nosec B603
```

Ruff (0.15.20)

[error] `subprocess` call: check for execution of untrusted input

(S603)